File: solutions/167.two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.js
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (numbers, target) => {
  let set = new Set()

  for (let i = 0; i < numbers.length; i++) {
    if (set.has(numbers[i])) {
      return [i+1, numbers.indexOf(target - numbers[i])+1].sort((a,b)=>a-b)
    }

    set.add(target - numbers[i])
  }
}

File: solutions/167.two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        l = 0
        r = len(numbers) - 1

        while l < r:
            sum = numbers[l] + numbers[r]
            if sum < target:
                l += 1
            elif sum > target:
                r -= 1
            else:
                return [l +  1, r + 1]
